exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/UpdateBuilder.kt

```diff
@@ -24,32 +24,32 @@ abstract class UpdateBuilder<out T>(type: StatementType, targets: List<Table>) :
     }
 
     @JvmName("setWithEntityIdExpression")
-    operator fun <S, ID : EntityID<S>, E : Expression<S>> set(column: Column<ID>, value: E) {
+    operator fun <S : Comparable<S>> set(column: Column<out EntityID<S>?>, value: Expression<out S?>) {
         require(!values.containsKey(column)) { "$column is already initialized" }
         column.columnType.validateValueBeforeUpdate(value)
         values[column] = value
     }
 
     @JvmName("setWithEntityIdValue")
-    operator fun <S : Comparable<S>, ID : EntityID<S>, E : S?> set(column: Column<ID>, value: E) {
+    operator fun <S : Comparable<S>> set(column: Column<out EntityID<S>?>, value: S?) {
```

Comment:
This is the key signature change to support `it[nullableIntRef] = null as Int?`. The rest (e.g. `out`) is done to keep existing tests pass, and make code consistent.